ext/fg/js/frontend.js

```diff
@@ -230,7 +230,7 @@ class Frontend extends TextScanner {
         const searchText = textSource.text();
         if (searchText.length === 0) { return null; }
 
-        const definitions = await apiKanjiFind(searchText, this.getOptionsContext());
+        const definitions = await apiKanjiFind(searchText, await this.getOptionsContext());
```

Comment:
There is now a redunant `await this.getOptionsContext()` when the following happens:

```js
results = (
    await this.findTerms(textSource) ||
    await this.findKanji(textSource)
);
```

Could instead `await this.getOptionsContext()` once inside of `onSearchSource` and pass it to both functions. It could also be passed to the `showContent` and `_showPopupContent` calls, partially eliminating the need for `getOptionsContextCached`. `_updatePopupPosition` could them be updated to `await this.getOptionsContext()` also, completely eliminating the need for `getOptionsContextCached`.

I don't think `Frontend.showContent` is used externally, so I don't think there would be a concern with adding an `optionsContext` parameter to it. `showContent` could then also use the `url` from the context instead of `this._url`, which may remove the need for that field (I think it's only used for the cached version?).